pyworkflow/protocol/protocol.py

```diff
@@ -2569,11 +2583,28 @@ def stepsGeneratorStep(self):
         """
         pass
 
-    def _validateThreads(self, messages:list):
+    def _getStreamingSleepOnWait(self):
+        """ Retrieves the configured sleep duration for waiting during streaming.
+            Returns:
+            - int: The sleep duration in seconds during streaming wait.
+            """
+        return self.getAttributeValue('streamingSleepOnWait', 0)
+
+    def _streamingSleepOnWait(self):
+        """ This method should be used by protocols that want to sleep
+        when there is not more work to do.
+        """
+        sleepOnWait = self._getStreamingSleepOnWait()
+        if sleepOnWait > 0:
+            self.info("Not much work to do now, sleeping %s seconds." % sleepOnWait)
```

Comment:
Maybe bettter this message?

"Waiting %s now before checking again for new input" % sleepOnWait